### PR TITLE
[Issue 962] Add VPC flow logs

### LINKED ISSUE #962: [Task]: VPC Security: Pass Security Hub Checks
### Summary

EC2 is being used to host nessus and burpsuite scanning tools. A number of EC2 checks in Security Hub are not passing. The suggested steps should be taken to remediate this to the checks pass. The checks are:

- [ ] [Amazon EC2 should be configured to use VPC endpoints that are created for the Amazon EC2 service](https://us-east-1.console.aws.amazon.com/securityhub/home?region=us-east-1#/standards/nist-800-53-5.0.0/EC2.10)
- [ ] [VPC flow logging should be enabled in all VPCs](https://us-east-1.console.aws.amazon.com/securityhub/home?region=us-east-1#/standards/nist-800-53-5.0.0/EC2.6)

### Acceptance criteria

- [ ] EC2 checks passing

### PR DESCRIPTION
**Pull request by coilysiren**
## Summary

Partially resolves #962

## Changes proposed

Adds a VPC flow logs cloudwatch logs group

## Context for reviewers

Most of this configuration is from here: https://github.com/terraform-aws-modules/terraform-aws-vpc/blob/master/examples/vpc-flow-logs/main.tf

## Testing

Tested in staging

image

### CODE REVIEW

File: infra/modules/network/flow-logs.tf [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+# docs: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/cloudwatch_log_group
+resource "aws_cloudwatch_log_group" "flow_log" {
+  name = "vpc-flow-logs-to-cloudwatch-${random_pet.this.id}"
+
+  # Conservatively retain logs for 5 years.
+  # Looser requirements may allow shorter retention periods
+  retention_in_days = 1827
+
+  # TODO(https://github.com/navapbc/template-infra/issues/164) Encrypt with customer managed KMS key
+  # checkov:skip=CKV_AWS_158:Encrypt service logs with customer key in future work
```
Comment (coilysiren) [inline infra/modules/network/flow-logs.tf:18]: Copied from elsewhere in this Github repo

File: infra/modules/network/flow-logs.tf [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    actions = [
+      "logs:CreateLogGroup",
+      "logs:CreateLogStream",
+      "logs:PutLogEvents",
+      "logs:DescribeLogGroups",
+      "logs:DescribeLogStreams",
+    ]
+
+    # checkov:skip=CKV_AWS_111:This is literally the AWS reccomended way to do this. Also its just logs.
+    resources = ["*"]
```
Comment (coilysiren) [inline infra/modules/network/flow-logs.tf:67]: This should be a standard pattern for logs. It's just logs, I'm not concerned about them getting hijacked and writing to the wrong place.

**Comment by coilysiren** [comment]
Ah wait I missed something... small commit incoming

**Comment by coilysiren** [comment]
Nevermind we're good 👍🏼

File: infra/modules/network/flow-logs.tf [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+# https://github.com/terraform-aws-modules/terraform-aws-vpc/blob/master/examples/vpc-flow-logs/main.tf
+
+# https://registry.terraform.io/providers/hashicorp/random/latest/docs/resources/pet
+resource "random_pet" "this" {
+  length = 3
+}
+
+# docs: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/cloudwatch_log_group
+resource "aws_cloudwatch_log_group" "flow_log" {
+  name = "vpc-flow-logs-to-cloudwatch-${random_pet.this.id}"
```
Comment (jamesbursa) [inline infra/modules/network/flow-logs.tf:11]: Why do we need the name randomized? Would using `name_prefix` instead of `name` be another way?

Comment (coilysiren) [reply infra/modules/network/flow-logs.tf:11]: This was a copy paste - you're right that we don't need this randomized per-se. I'll check if there's a name prefix...

Comment (coilysiren) [reply infra/modules/network/flow-logs.tf]: Updated! https://github.com/HHS/simpler-grants-gov/pull/1089/commits/3a090cce230ad7701a5e604c6b02fa90d1e56c83

**Review by jamesbursa — APPROVED** [review]

**Pull request merged**